File: apps/StaffApp.Android/app/src/main/java/com/acls/staff/domain/model/Staff.kt
package com.acls.staff.domain.model

data class Staff(
    val staffMemberId: Int,
    val userId: Int,
    val fullName: String,
    val jobTitle: String?,
    val skills: List<String>,
    val availability: String,
    val averageRating: Double?,
    val lastAssignedAt: String?,
    val activeAssignments: List<Complaint>
)

File: packages/api-contracts/src/staff.ts
// Staff request/response types mirroring ACLS Application DTOs

import type { ComplaintSummaryDto } from "./complaints";

export interface StaffMemberDto {
  staffMemberId: number;
  userId: number;
  fullName: string;
  jobTitle: string | null;
  skills: string[];
  availability: string;
  averageRating: number | null;
  lastAssignedAt: string | null;
}

export interface StaffMemberWithAssignmentsDto extends StaffMemberDto {
  activeAssignments: ComplaintSummaryDto[];
}

export interface UpdateAvailabilityRequest {
  availability: string;
}
